### PR TITLE
retrieve project from the name

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/13307

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
datacatalog: fixed the import failure when the `project` is different from the default on `google_data_catalog_taxonomy`
```

### CODE REVIEW

File: mmv1/templates/terraform/custom_import/data_catalog_taxonomy.go.erb [hunk 1]
```diff
@@ -8,4 +8,10 @@
     name := d.Get("name").(string)
     d.SetId(name)
 
+    re := regexp.MustCompile("projects/(.+)/(?:locations|regions)/(.+)/taxonomies/(.+)")
+	if matches := re.FindStringSubmatch(name); matches != nil {
+		d.Set("project", matches[1])
+        d.Set("locations", matches[2])
+	}
```
Comment (SarahFrench) [inline mmv1/templates/terraform/custom_import/data_catalog_taxonomy.go.erb:15]:
```suggestion
    if matches := re.FindStringSubmatch(name); matches != nil {
        d.Set("project", matches[1])
        d.Set("locations", matches[2])
    }
```

**Review by SarahFrench — APPROVED** [review]
Happy new year! This PR looks good to me, and I like how you've used `FindStringSubmatch` to get values instead of just splitting the name string by `/`.

I've included a suggested change for a nit about indentation, but I'll approve as it's not blocking

**Review by SarahFrench — APPROVED** [review]
I re-requested myself for review because I learned some more about the `parseImportId` function from Riley and I wanted to make sure I understood this problem fully.

From my understanding, the import code usually made for generated resources would pull in project names ok, but this resource's names can include slashes so we need to use custom import code, right?

In that case, re-approved!

**Review by SarahFrench — APPROVED** [review]

**Comment by edwardmedia** [comment]
@SarahFrench usually name is short while id is long. In this case, the name is long and no other fields (like project and locations) are available which are needed for building the url for read call.  I looked up the attributes and it appears we have to parse the name to obtain project and locations. Fortunately there is already import custom code which makes the fix a little easier.

**Comment by c2thorn** [comment]
This appears to be causing a crash in VCR runs @edwardmedia

**Pull request merged**